pandas/core/generic.py

```diff
@@ -7922,34 +7922,97 @@ def last(self, offset):
-            * top: smallest rank if ascending
-            * bottom: smallest rank if descending
-        ascending : boolean, default True
-            False for ranks by high (1) to low (N)
-        pct : boolean, default False
-            Computes percentage rank of data
+        numeric_only : bool, optional
+            For DataFrame objects, rank only numeric columns if set to True.
+        na_option : {'keep', 'top', 'bottom'}, default 'keep'
+            How to rank NaN values:
+
+            * keep: assign NaN rank to NaN values
+            * top: assign smallest rank to NaN values if ascending
+            * bottom: assign highest rank to NaN values if ascending
+        ascending : bool, default True
+            False for ranks by high (1) to low (N).
+        pct : bool, default False
+            Computes percentage rank of data.
 
         Returns
         -------
         ranks : same type as caller
+            Return a Series or DataFrame with data ranks as values
+
+        See Also
+        --------
+        pandas.core.groupby.GroupBy.rank : Rank of values within each group.
+
+        Examples
+        --------
+        The default behavior returns average ranks of every column
+
+        >>> df = pd.DataFrame(data={'Customer': [1, 2, 3, 4, 5],
+        ...                         'Tot_Spend': [12, 20, 20, 18, 16]})
+        >>> df.rank()
+           Customer  Tot_Spend
+        0       1.0        1.0
+        1       2.0        4.5
+        2       3.0        4.5
+        3       4.0        3.0
+        4       5.0        2.0
+
+        The following examples show how rank behaves with every different
+        method and when setting pct = True
+
+        >>> df['default_rank'] = df['Tot_Spend'].rank()
+        >>> df['min_rank'] = df['Tot_Spend'].rank(method='min')
+        >>> df['max_rank'] = df['Tot_Spend'].rank(method='max')
+        >>> df['dense_rank'] = df['Tot_Spend'].rank(method='dense')
+        >>> df[['Tot_Spend', 'default_rank', 'min_rank', 'max_rank',
+        ...     'dense_rank']]
+           Tot_Spend  default_rank  min_rank  max_rank  dense_rank
+        0         12           1.0       1.0       1.0         1.0
+        1         20           4.5       4.0       5.0         4.0
+        2         20           4.5       4.0       5.0         4.0
+        3         18           3.0       3.0       3.0         3.0
+        4         16           2.0       2.0       2.0         2.0
+
+        >>> df['default_rank'] = df['Tot_Spend'].rank()
+        >>> df['pct_rank'] = df['Tot_Spend'].rank(pct=True)
+        >>> df[['Tot_Spend', 'default_rank', 'pct_rank']]
+           Tot_Spend  default_rank  pct_rank
+        0         12           1.0       0.2
+        1         20           4.5       0.9
+        2         20           4.5       0.9
+        3         18           3.0       0.6
+        4         16           2.0       0.4
+
+        The following example shows how rank behaves with NAs
```

Comment:
Can you add nans to the first example, so we can remove this part, and things are a bit more compact.